src/main/java/org/mifos/connector/channel/interceptor/IdInterceptor.java

```diff
@@ -0,0 +1,144 @@
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+            throws Exception {
+
+        if (redisIdempotencyEnabled) {
+           Set<String> listParts = new HashSet<>(Arrays.asList(apiList.split(",")));
+              if (listParts.contains(request.getRequestURI())) {
+                    return checkIdempotency(request, response);
+               } else
+                   logger.debug("redis idempotency is disabled for this api,proceeding with request");
+                  return true;
+            }
+        else
+            logger.debug("redis idempotency is disabled,proceeding with request");
+        return true;
+    }
+
+    private boolean checkIdempotency(HttpServletRequest request, HttpServletResponse response) {
+        PhErrorDTO errorDTO = null;
+        String correlationId = request.getHeader(CLIENTCORRELATIONID);
+        if(correlationId.isBlank()) {
+            logger.debug("X-CorrelationID is missing,attaching error dto to response");
+            errorDTO = new PhErrorDTO.PhErrorDTOBuilder(PaymentHubError.MandatoryValueNotSupplied)
+                    .addErrorParameter(CLIENTCORRELATIONID, "X-CorrelationID")
+                    .developerMessage("Pass the valid header value for " + CLIENTCORRELATIONID)
+                    .build();
+            createError(response, errorDTO);
+            return false;
+        }
+        String key = createRedisKey(request.getHeader("Platform-TenantId"),
+                correlationId,request.getRequestURI(),keyFormat);
+        logger.debug("X-CorrelationID value is {}", correlationId);
+            if (Boolean.TRUE.equals(redisTemplate.hasKey(correlationId))) {
+                logger.debug("X-CorrelationID exists already in redis,attaching error dto to response");
+                errorDTO = new PhErrorDTO.PhErrorDTOBuilder(PaymentHubError.TransactionExistsError)
+                        .addErrorParameter(CLIENTCORRELATIONID, correlationId)
+                        .developerMessage("Pass the valid header value for " + CLIENTCORRELATIONID)
+                        .build();
+                createError(response, errorDTO);
+                response.setHeader(CLIENTCORRELATIONID, correlationId);
+                return false;
+            } else {
+                logger.debug("X-CorrelationID does not exist in redis,proceeding with request");
+                redisTemplate.opsForValue().set(key, correlationId, cacheRetencyDuration,
+                        TimeUnit.DAYS);
+                return true;
+            }
+
+    }
+
+    private String createRedisKey(String tenant, String correlationId, String requestURI, String keyFormat) {
+       keyFormat = keyFormat.replace("{tenant}", tenant).replace("{clientCorrelationId}", correlationId)
+               .replace("{api}", requestURI);
```

Comment:
if the format is missing the api , will this throw error?

Reply:
checking